protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1276,15 +1279,26 @@ func renderServices(services []*descriptor.Service, paths *openapiPathsObject, r
 							if err != nil {
 								return err
 							}
+
 							if len(b.PathParams) == 0 {
 								if err := schema.setRefFromFQN(meth.RequestType.FQMN(), reg); err != nil {
 									return err
 								}
 								desc = messageSchema.Description
 							} else {
-								schema = messageSchema
-								if schema.Properties == nil || len(*schema.Properties) == 0 {
-									grpclog.Warningf("created a body with 0 properties in the message, this might be unintended: %s", *meth.RequestType)
+								if messageSchema.Properties != nil && len(*messageSchema.Properties) != 0 {
+									// Combine the proto filepath and the method name for the definition's name
+									protoFilePath := lastFile.GetName()
+									protoFilePath = strings.TrimSuffix(protoFilePath, filepath.Ext(protoFilePath))
+									protoFilePath = nonAlphanumeric.ReplaceAllString(protoFilePath, "_")
+
+									defName := fmt.Sprintf("%s_%sBody", protoFilePath, meth.GetName())
+
+									schema.Ref = fmt.Sprintf("#/definitions/%s", defName)
+									defs[defName] = messageSchema
+								} else {
+									schema = nil
+									grpclog.Warningf("skipping a body with 0 properties in the message, this might be unintended: %s", *meth.RequestType)
```

Comment:
Hm, was the old behavior to just include an `object` in these? It does feel right to just skip it as there should be no body parameters, but I really hope we don't break users.

Reply:
Yes, the old behavior would just include an empty body. The result makes more sense but we can drop it if you think this is too risky.

Reply:
I'd say we should keep the old behavior as much as possible, so please change this back.

Reply:
ok, removed